stdlib/public/runtime/Mutex.cpp

```diff
@@ -0,0 +1,98 @@
+//===--- Mutex.cpp - Lockables --------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Mutex, Condition, and Scoped lock abstactions for use in Swift runtime.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Mutex.h"
+
+#include "swift/Runtime/Debug.h"
+#include <errno.h>
+#include <stdlib.h>
+
+using namespace swift;
+
+#define reportError(PThreadFunction)                                           \
+  {                                                                            \
```

Comment:
The usual way to define multi-statement macros is to use `do { ... } while false`, and allow the semicolon from the call site to end the statement.